packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx

```diff
@@ -1,427 +1,187 @@
 import * as React from 'react';
 import {
   BaseComponent,
-  autobind,
-  buttonProperties,
-  anchorProperties,
   css,
-  divProperties,
   getId,
-  getNativeProps
+  autobind,
+  assign
 } from '../../Utilities';
-import { ICommandBar, ICommandBarProps } from './CommandBar.Props';
-import { FocusZone, FocusZoneDirection } from '../../FocusZone';
-import { ContextualMenu, IContextualMenuProps, IContextualMenuItem, hasSubmenuItems } from '../../ContextualMenu';
-import { DirectionalHint } from '../../common/DirectionalHint';
-import {
-  Icon,
-  IconName,
-  IIconProps
-} from '../../Icon';
-import { FontClassNames } from '../../Styling';
+import { ICommandBar, ICommandBarProps, ICommandBarItemProps } from './CommandBar.Props';
+import { CommandButton } from '../../Button';
+import { OverflowSet } from '../../OverflowSet';
+import { FocusZone } from '../../FocusZone';
+import { ResizeGroup } from '../../ResizeGroup';
+import { TooltipHost } from '../../Tooltip';
 import * as stylesImport from './CommandBar.scss';
 const styles: any = stylesImport;
 
-const OVERFLOW_KEY = 'overflow';
-const OVERFLOW_WIDTH = 41.5;
-
-export interface ICommandBarState {
-  renderedItems?: IContextualMenuItem[];
-  renderedOverflowItems?: IContextualMenuItem[];
-  expandedMenuItemKey?: string;
-  expandedMenuId?: string;
-  contextualMenuProps?: IContextualMenuProps;
-  contextualMenuTarget?: HTMLElement;
-  renderedFarItems?: IContextualMenuItem[];
+export interface ICommandBarData {
+  primaryItems: ICommandBarItemProps[];
+  overflowItems: ICommandBarItemProps[];
+  farItems: ICommandBarItemProps[] | undefined;
+  originalOverflowItems: ICommandBarItemProps[];
```

Comment:
yeah, if I say "put these 3 in the overflow to start", we don't want to accidentally pull them out when onGrowData.